src/inc/quic_platform_linux.h

```diff
@@ -633,6 +633,28 @@ typedef struct QUIC_THREAD_CONFIG {
     void* Context;
 } QUIC_THREAD_CONFIG;
 
+#ifdef QUIC_USE_CUSTOM_THREAD_CONTEXT
+
+//
+// Extension point that allows additional platform specific logic to be executed
+// for every thread created. The platform must define QUIC_USE_CUSTOM_THREAD_CONTEXT
+// and implement the QuicThreadCustomStart function. QuicThreadCustomStart MUST
+// call the Callback passed in.
```

Comment:
This needs to also document that the implementation must free `CustomContext` after exiting. Since they have access to platform, its probably a good idea to use QUIC_ALLOC and QUIC_FREE instead of malloc and free. We can make a tag, and document what the free call needs to be.